cpuset.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * cpuset.h -- CPU affinity.
+ *
+ * Copyright (c) 2020, NLnet Labs. All rights reserved.
+ *
+ * See LICENSE for the license.
+ *
+ */
+
+#if defined(__linux__)
+#define _GNU_SOURCE
+#include <sched.h>
+#endif /* __linux__ */
+
+#include <assert.h>
+#include <limits.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include "cpuset.h"
+#include "config.h"
```

Comment:
need config.h at the start of the includes because of defines for different include standards in config.h.

Reply:
If you do this, the _GNU_SOURCE define may then already be defined in config.h, and then you have to check with ifdef for that before you define it for sched.h above.

Reply:
Done